Sources/securekeystore/Biometrics/BiometricsImpl.swift

```diff
@@ -92,21 +86,26 @@ class BiometricsImpl: BiometricsProtocol {
         return context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error)
     }
     
-    // Private method to check if authentication is required based on the delay
-    private func isAuthenticationRequired() -> Bool {
+    // Private method to check if authentication is required based on the key type and its timeout
+    private func isAuthenticationRequired(forKeyType keyType: String) -> Bool {
         return authQueue.sync {
-            guard let lastAuth = lastSuccessfulAuthTime else {
-                return true // Require authentication if none has been recorded
+            guard let lastAuth: Date = self.lastSuccessfulAuthTime[keyType] else {
+                if(getAuthTimeOutForKeyType(keyType: keyType) > -1)
```

Comment:
extract `getAuthTimeOutForKeyType(keyType: keyType)` into a variable and use at line# 93, 99